model/src/main/java/io/paradoxical/cassieq/model/auth/SignatureGenerator.java

```diff
@@ -0,0 +1,43 @@
+package io.paradoxical.cassieq.model.auth;
+
+import com.godaddy.logging.Logger;
+import io.paradoxical.cassieq.model.accounts.AccountKey;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+import static com.godaddy.logging.LoggerFactory.getLogger;
+
+/**
+ * Signs paramters
+ */
+public interface SignatureGenerator {
```

Comment:
would prefer to keep hmac sha 256 out of the interface. can we have an abstract class `HmacSha256SignatureGenerator` which does this?